src/shared/types/global.d.ts

```diff
@@ -1,4 +1,4 @@
-export interface CustomWindow extends Window {
```

Comment:
Het is een goed idee om overal waar je window aanspreekt dit eerst te casten naar een interface, zodat je perfect weet wat er op het globaal window gezet wordt en welke types die objecten hebben.

```
(window as CustomWindow).APP_INFO = {
	mode: process.env.NODE_ENV,
	version: pkg.version,
};
```
Laat toe om typos te vangen:
![image](https://user-images.githubusercontent.com/1710840/67661342-436ef580-f961-11e9-98b4-45235d56def7.png)

Je kan dit ook extenden van de native Window interface:

```
export interface CustomWindow extends Window {
	_ENV_: {
		PROXY_URL: string;
		FLOW_PLAYER_TOKEN: string;
		FLOW_PLAYER_ID: string;
		PORT: string;
		NODE_ENV: string;
	};
	APP_INFO: {
		version: string;
		mode: 'development' | 'production' | 'test';
	};
}
```

Reply:
Typescript pikt dit vanzelf op als je gwn properties toevoegt aan de Window interface? Ze raden dit zelfs aan in hun eigen docs, dan moet die interface niet meer geëxport worden.
De reden waarom ik dit heb gedaan was omdat ik het volgende kreeg:
![image](https://user-images.githubusercontent.com/16118473/67665587-491d0900-f96a-11e9-9783-cceea02af121.png)

Reply:
Also, typechecking zoals je hierboven aanhaalt werkt ook nog:
![image](https://user-images.githubusercontent.com/16118473/67666844-ebd68700-f96c-11e9-849c-83061ff1a140.png)

Reply:
nee als de typechecking werkt dan is het geen probleem